app/models/course/assessment/submission.rb

```diff
@@ -12,6 +12,8 @@ class Course::Assessment::Submission < ApplicationRecord
 
   workflow do
     state :attempting do
+      event :finalise, transitions_to: :published,
+                       if: proc { |submission| submission.assessment.questions.empty? }
```

Comment:
Suggest we add a TODO here to wait till Workflow is upgraded to support symbols. For now, I don't expect it to be updated soon since it doesn't seem to be actively maintained. Can take this into a separate discussion on what to do later on.